src/ui/tabs/NestedTab.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from '../styles/overlay.module.css';
+
+function NestedTab(props) {
```

Comment:
Ok, I think I see what's happening here. It took me a while to figure out what this component does, which is usually a sign that it should be reworked a bit.

From what I gather, this is sort of a "progressive tabs" component, where a parent tab renders a child but the child is only rendered when it is active, in which case it takes precedence.

In your specific case, you render a "Select Character" tab, and the user must select a profile. Then the active tab will be set to the child "Character Sheet" tab, which will bring it to the forefront. Clicking the "Select Character" tab will make that one active again and hide the "Character Sheet" tab.

The funny thing is that you've basically implemented a rudimentary version of what is called a "router". A router manages navigation through an app. The most widely used router for React applications is [react-router](https://reacttraining.com/react-router/). Routers are usually necessary so that the app route can be kept synchronized with the browser URL. In an extension that won't be necessary, so you can manage it similarly to the way you have it here.

This component is more generic than it needs to be. I would just move this into `Overlay` and base everything on whether a sheet has been selected:

```jsx
function Overlay() {
    const [sheet, setSheet] = useState(null);

    // this is ignoring some other changes I'm going to suggest later about promises
    const onCharacterReady = (characterSheetPromise) => {
        characterSheetPromise.then((sheetData) => {
            setSheet(sheetData);
        });
    };

    return (
        <div className={`globalOverlay ${styles.Overlay}`}>
            <span className={styles.NestedTabHeader}>
                {/* This button can probably be a separate component */}
                <button ...>Select Character</button>
                {sheet && (
                    <span>{'>'}</span>
                    <button ...>Character Sheet</button>
                )}
            </span>
            {!sheet ? <ProfileSelection onCharacterReady={onCharacterReady} />}
                : <CharacterSheet sheet={sheet} />}
        </div>
    );
}
```